src/material/core/common-behaviors/error-state.ts

```diff
@@ -45,17 +49,20 @@ export function mixinErrorState<T extends AbstractConstructor<HasErrorState>>(ba
 export function mixinErrorState<T extends Constructor<HasErrorState>>(base: T):
   CanUpdateErrorStateCtor & T {
   return class extends base {
+    // This class member exists as an interop with `MatFormFieldControl` which expects
+    // a public `stateChanges` observable to emit whenever the form field should be updated.
+    // The description is not specifically mentioning the error state, as classes using this
+    // mixin can/should emit an event in other cases too.
+    /** Emits whenever the component state changes. */
+    readonly stateChanges = new Subject<void>();
```

Comment:
I've added the internal comment here because I was actually surprised about `stateChanges` being defined here. I've also wondered on whether it should be made docs-private or not. I'm unsure on whether we should move this out of the mixin (as a cleanup) and require the mixin base classes to define the `stateChanges` property, but it looks like this mixin is commonly used within form-fields.. I kept it public because there are other cases where we have `stateChanges` intentionally showing-up in the docs.

Reply:
Now that you mention it, it does sound weird that this is initialized in the mixin. Maybe it should be moved to all the places that apply the mixin instead? The interface should enforce that it's always defined.

Reply:
Yeah, I agree, I was surprised too and left this comment to start the conversation.

Reply:
@crisbeto I've went ahead and just made the changes as part of this PR.